azkaban-common/src/main/java/azkaban/metrics/MetricsManager.java

```diff
@@ -0,0 +1,120 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.metrics;
+
+import azkaban.AzkabanConstants;
+import azkaban.utils.Props;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ConsoleReporter;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.ThreadStatesGaugeSet;
+
+import org.apache.log4j.Logger;
+
+import java.lang.reflect.Constructor;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The singleton class, MetricsManager, is the place to have MetricRegistry and
+ * ConsoleReporter in this class. Also, web servers and executors can call {@link #startReporting(Props)}
+ * to start reporting AZ metrics to remote metrics server.
+ */
+public enum MetricsManager {
+  INSTANCE;
+
+  private final MetricRegistry registry        = new MetricRegistry();
+  private ConsoleReporter consoleReporter      = null;
+  private static final Logger logger = Logger.getLogger(MetricsManager.class);
+
+  /**
+   * Constructor is eaagerly called when this class is loaded.
+   */
+  private MetricsManager() {
+    registry.register("MEMORY_Gauge", new MemoryUsageGaugeSet());
+    registry.register("GC_Gauge", new GarbageCollectorMetricSet());
+    registry.register("Thread_State_Gauge", new ThreadStatesGaugeSet());
+  }
+  /**
+   * Return the Metrics registry.
+   *
+   * @return the single {@code MetricRegistry} used for all of Az Metrics
+   *         monitoring
+   */
+  public MetricRegistry getRegistry() {
+    return registry;
+  }
+
+  /**
+   * reporting metrics to remote metrics collector.
+   * Note: this method must be synchronized, since both web server and executor
+   * will call it during initialization.
+   */
+  public synchronized void startReporting(Props props) {
+    String metricsReporterClassName = props.get(AzkabanConstants.CUSTOM_METRICS_REPORTER_CLASS_NAME);
+    String metricsServerURL = props.get(AzkabanConstants.METRICS_SERVER_URL);
+    if (metricsReporterClassName != null && metricsServerURL != null) {
+      try {
+        logger.info("metricsReporterClassName: " + metricsReporterClassName);
+        Class metricsClass = Class.forName(metricsReporterClassName);
+
+        Constructor[] constructors =
+            metricsClass.getConstructors();
+        constructors[0].newInstance(registry, metricsServerURL);
+
+      } catch (Exception e) {
+        logger.error("Encountered error while loading and instantiating "
+            + metricsReporterClassName, e);
+        throw new IllegalStateException(
+            "Encountered error while loading and instantiating "
+                + metricsReporterClassName, e);
+      }
+    } else {
+      logger.info("No value for property: "
```

Comment:
Checked the logic. since METRICS_ENABLE already passed in this function, we should use Error. changed.